### DISCUSSION TITLE
Seemingly random 404s on SSG

### DISCUSSION THREAD

**Discussion by shotor** [vercel/next.js#39914] [Help]
I have a weird issue I'm debugging. Looking to see if anyone else has this problem.

We run 2 instances of NextJS on AWS behind a loadbalancer. After an unknown amount of time, one of the instances starts giving 404s for the url: `/_next/data/{build-id}/en.json` (en being the locale). There's 1 codepath on this page that can return a 404 and there we set revalidate to 1.

The weird thing is that the cache-control header is set to: `s-maxage=31536000, stale-while-revalidate`. When returning the 404. The other instance happily returns a 200 with the data.

This breaks our homepage periodically. We're using Next `12.2.1`. I've been unable to reliably reproduce because it happens seemingly randomly. The logs don't indicate any problems.

Questions:

- Is there a way to log more information about static generation? I'd like to see whether a page is being regenerated or not, and what the status was.
- Could this be conflicting: We have an index.ts as well as a [slug].ts. In the chrome network logs I sometimes see this call being made: `/_next/data/{build-id}/en.json?slug=en` which seems to suggest it's also being recognized as a dynamic route. But when you manually call that endpoint you get the normal data

**Comment by kmanimurugan** [comment]
I am also facing the same issue
